reconcile/rhidp/ocm_oidc_idp/base.py

```diff
@@ -21,48 +16,59 @@
     RhIdpOCMOidcIdpReconcileErrorCounter,
 )
 from reconcile.utils import metrics
+from reconcile.utils.differ import diff_iterables
 from reconcile.utils.keycloak import SSOClient
-from reconcile.utils.ocm import OCMMap
+from reconcile.utils.ocm.base import (
+    OCMOIdentityProvider,
+    OCMOIdentityProviderGithub,
+    OCMOIdentityProviderOidc,
+    OCMOIdentityProviderOidcOpenId,
+)
+from reconcile.utils.ocm.identity_providers import (
+    add_identity_provider,
+    delete_identity_provider,
+    get_identity_providers,
+    update_identity_provider,
+)
+from reconcile.utils.ocm_base_client import OCMBaseClient
 from reconcile.utils.secret_reader import SecretReaderBase
 
-DEFAULT_EMAIL_CLAIMS: list[str] = ["email"]
-DEFAULT_NAME_CLAIMS: list[str] = ["name"]
-DEFAULT_USERNAME_CLAIMS: list[str] = ["preferred_username"]
-DEFAULT_GROUPS_CLAIMS: list[str] = []
+
+class IDPState(BaseModel):
+    cluster: Cluster
+    idp: OCMOIdentityProvider | OCMOIdentityProviderOidc | OCMOIdentityProviderGithub
+
+    def __eq__(self, __value: object) -> bool:
+        if not isinstance(__value, IDPState):
+            raise NotImplementedError("Cannot compare to non IDPState objects.")
+        return self.idp == __value.idp
 
 
 def run(
     integration_name: str,
     ocm_environment: str,
-    clusters: Iterable[ClusterV1],
+    clusters: Iterable[Cluster],
     secret_reader: SecretReaderBase,
+    ocm_api: OCMBaseClient,
     vault_input_path: str,
     dry_run: bool,
-    managed_idps: Optional[list[str]] = None,
+    managed_idps: list[str] | None = None,
```

Comment:
Is this the new preferred typing for optional field?

Reply:
This came with Python 3.11, and I like it more :)